src/RepoDriver.sol

```diff
@@ -0,0 +1,530 @@
+    /// The driver will cover the fee in Link that must be paid to the operator.
+    /// If you want to cover the fee yourself, use `onTokenTransfer`.
+    ///
+    /// The repository must contain a `FUNDING.json` file in the project root in the default branch.
+    /// The file must be a valid JSON with arbitrary data, but it must contain the owner address
+    /// as a hexadecimal string under `drips` -> `<CHAIN NAME>` -> `ownedBy`, a minimal example:
+    /// `{ "drips": { "ethereum": { "ownedBy": "0x0123456789abcDEF0123456789abCDef01234567" } } }`.
+    /// If the operator can't read the owner when processing the update request,
+    /// it ignores the request and no change to the user ownership is made.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// For GitHub and GitLab it must follow the `user_name/repository_name` structure
+    /// and it must be formatted identically as in the repository's URL,
+    /// including the case of each letter and special characters being removed.
+    /// @return userId The ID of the user.
+    function requestUpdateOwner(Forge forge, bytes memory name)
+        public
+        whenNotPaused
+        returns (uint256 userId)
+    {
+        uint256 fee = _repoDriverAnyApiStorage().defaultFee;
+        require(linkToken.balanceOf(address(this)) >= fee, "Link balance too low");
+        return _requestUpdateOwner(forge, name, fee);
+    }
+
+    /// @notice The function called when receiving funds from ERC-677 `transferAndCall`.
+    /// Only supports receiving Link tokens, callable only by the Link token smart contract.
+    /// The only supported usage is requesting user ownership updates,
+    /// the transferred tokens are then used for paying the AnyApi operator fee,
+    /// see `requestUpdateOwner` for more details.
+    /// The received tokens are never refunded, so make sure that
+    /// the amount isn't too low to cover the fee, isn't too high and wasteful,
+    /// and the repository's content is valid so its ownership can be verified.
+    /// @param amount The transferred amount, it will be used as the AnyApi operator fee.
+    /// @param data The `transferAndCall` payload.
+    /// It must be a valid ABI-encoded calldata for `requestUpdateOwner`.
+    /// The call parameters will be used the same way as when calling `requestUpdateOwner`,
+    /// to determine which user's ownership update is requested.
+    function onTokenTransfer(address, /* sender */ uint256 amount, bytes calldata data)
+        public
+        whenNotPaused
+    {
+        require(msg.sender == address(linkToken), "Callable only by the Link token");
+        require(data.length >= 4, "Data not a valid calldata");
+        require(bytes4(data[:4]) == this.requestUpdateOwner.selector, "Data not requestUpdateOwner");
+        (Forge forge, bytes memory name) = abi.decode(data[4:], (Forge, bytes));
+        _requestUpdateOwner(forge, name, amount);
```

Comment:
We let the user here pass an `amount` unequal to `fee`?  Maybe we should `revert` if the fee is not sufficient or use the `fee` variable.
On the other hand currently, the funds locked in the contract can be used by just calling `requestUpdateOwner`.

Reply:
Yes, accepting a custom fee is on purpose, it provides a fallback in case the fees change and we don't update the configuration. It's really hard to tell whether the fee is sufficient, there's no information available on-chain.

Just to be clear, this code is called after receiving `amount` Link tokens from the user, we're passing 100% of the received funds as a fee.

Reply:
My assumption here is if `fee >= defaultFee` it should be sufficient. However, we could also allow it to pass 100%. 
Then, it could be used in case the configured `defaultFee` is not sufficient anymore.

Reply:
The default fee can also be too high, which would force our users to pay unreasonable fees unless we lower the default fee. I think that putting a hard limit on the range of fees is an unnecessary complication, it maybe protects from typos in the transferred amounts, but it brings weird restrictions to the API. If the user wants to pay X Link for the verification, let them. We may put some sanity checks in the app to avoid stupid mistakes.